### ISSUE TITLE
Trouble following packaging libraries tutorial

### ISSUE THREAD

**Issue by jonhue** [pypa/packaging-problems#258]
Thank you for providing feedback on Python packaging!

To help us help you, please fill out as much of the following as you can. If a question is not relevant, feel free to skip it.

1. What is your operating system and version?

Ubuntu 18.04

2. What is your Python version?

Python 3.6.7

3. What version of pip do you have?

pip 9.0.1

4. Could you describe your issue in as much detail as possible?

I was following [this](https://packaging.python.org/tutorials/packaging-projects/) guide to locally package install my project. I installed the latest versions of `setuptools` and `wheel` (`python3 -m pip install --user --upgrade setuptools wheel`) and then tried to generate distribution archives. This step is where my problem lies. When running `sudo python3 setup.py sdist bdist_wheel`, I get a lot of `Copying ...` and then

```
running install_scripts 
error: [('build/bdist.linux-x86_64/wheel/few-0.1.0a0-py3.6.egg-info', 'build/bdist.linux-x86_64/wheel/few-0.1.0a0.dist-info', "[Errno 13] Permission denied: 'build/bdist.linux-x86_64/wheel/few-0.1.0a0.dist-info'")]
```

Do you have an idea what exactly the problem is?

**Comment by jonhue** [comment]
I was able to successfully run the same command a month ago. Don't know what changed though...

**Comment by dmtucker** [comment]
@jonhue Any chance you are running this on an NFS-mounted filesystem? Why are you using `sudo`?

**Comment by jonhue** [comment]
@dmtucker No, not using NFS. I used `sudo` because when not using it, I got an error saying that permission was denied trying to access a log file.

**Comment by CyrusNajmabadi** [comment]
I'm getting a very similar issue.

In our case, it's:

```
pipenv run python setup.py build bdist_wheel --universal

// a lot of copying

running install_scripts
error: [('build/bdist.linux-x86_64/wheel/pulumi-0.17.16.dev1559756548-py3.6.egg-info', 'build/bdist.linux-x86_64/wheel/pulumi-0.17.16.dev1559756548.dist-info', "[Errno 13] Permission denied: 'build/bdist.linux-x86_64/wheel/pulumi-0.17.16.dev1559756548.dist-info'")]
```

I'm not running sudo.  The `build/bdist.linux-x86_64/wheel/pulumi-0.17.16.dev1559756548.dist-info` directory exists and is fully owned by me.  I'm not on NFS.  

1. What is your operating system and version?

Ubuntu 18.04

2. What is your Python version?

Python 2.7.15rc1

3. What version of pip do you have?

pip 19.0.1

**Comment by rgillen** [comment]
Yup. Seems this is indeed broken... even running as full ROOT in home directory fails. Epic fail... :(

Ubuntu/Python 3.6.7/Pip 19.0.1

**Comment by pradyunsg** [comment]
Hey @pganssle, any idea what's happening here?

**Comment by pganssle** [comment]
@pradyunsg No idea. In the past I have found that most permissions issues are highly system-specific.

If possible, I recommend adding a `pyproject.toml` file and using `pep517.build` to build your sdist and wheels.

In any case, there does not seem to be enough information in this ticket to recreate the issue. Please consider crafting a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). I am guessing that the original poster's "few" package has a naming collision with [this project](https://pypi.org/project/FEW/), since the author seems to be different, and [pulumi](https://pypi.org/project/pulumi/) only seems to ship wheels and the github link points to a project written in Go, so I can't even try reproducing either of these problems.

@CyrusNajmabadi If you are involved in the "pulumi" project, you should also note that I think you have your `find_packages` misconfigured, because the latest `pulumi` wheel installs a package called "test" in addition to the package called "`pulumi`".

**Comment by rgillen** [comment]
Thanks for looking into this. I can provide a MRE in the next day or so (hopefully via a Docker image)

As it currently stands I can provide these additional details about the environment that produced the error...

I was attempting to build out a new developer laptop so at the time my system was rather 'clean'.

I DO NOT have Python2.x installed
Python 3.6.7 is ALSO installed for Windows

Ubuntu 18.04 via WSL on Windows 10 Enterprise, 1903.
`apt-get install -y python3`
`/usr/bin/python3 -m pip install --user --upgrade setuptools wheel`
`/usr/bin/python3 setup.py sdist --format gztar bdist_wheel`

**Comment by pganssle** [comment]
If I am correct that @jonhue was trying to build [this few-lang repo](https://github.com/few-lang/few), then I notice that he *also* erroneously is creating a package named `test`, because he's not using a `src` layout.

@rgillen Can you share the repo that this is failing for? I suspect that this permissions issue is arising because you're trying to build a repo called "test". Not sure why that would cause a problem, but it seems like a pretty big coincidence (and easy enough to fix).

**Comment by rgillen** [comment]
@pganssle - I am not trying to build a package named 'test', the package is named 'cogid'. It is a wrapper for boto3 Coginito Identity.

I was able to get the commands to work by running them out of my $HOME directory. Interestingly I tried this last night as ROOT and it didn't work (it was very late, I could have made an error in my test case).

I compared the permissions on the 2 folder structures and they are identical. The only difference is one is a WSL.config mount to `/e` and the other is the WSL internal path @ `\\wsl$\Ubuntu\home\svt`

So in summary...

**Fails**
 `w/CWD = /e/Git/Repos/pypi/cogid` (mapped via: wsl.config [automount] ROOT = /)


**Succeeds**
 `w/CWD = ~/cogid` (maps to: $\Ubuntu\home\svt\cogid)

**Comment by pganssle** [comment]
> I am not trying to build a package named 'test', the package is named 'cogid'. It is a wrapper for boto3 Coginito Identity.

I don't think you or anyone in this thread is *intentionally* trying to build a package called `test`. I know the other two people are doing it by accident. Try building a wheel and opening the wheel file (it's a zip file) - if you're not accidentally building any additional packages, you'll only see 2 folders, a `cogid` and a `cogid-XXX.dist-info` folder. If you see a folder called `test` or `examples` or `tests` or anything like that, you're accidentally building an extra package.

**Comment by Mimetis** [comment]
I have the exact same issue.
Trying to build a simple **whl** package results in the same kind of error.
I'm using **Ubuntu** through **WSL** on my Windows 10 machine (up to date)

My `setup.py` is pretty straightforward:

``` python
# -*- coding: utf-8 -*-

from setuptools import setup, find_packages

setup(name='southridge',
      version='0.9',
      description='Cleanse southridge datas',
      url='http://github.com/Mimetis',
      author='Sébastien Pertus',
      author_email='spertus@microsoft.com',
      license='MIT',
      packages=['southridge'],
      zip_safe=False)
```
and the result when I tried to build it is:

``` bash
spertus@MSI2019:/mnt/c/PROJECTS/DATA/sp/src$ sudo python3 setup.py bdist_wheel --universal
running bdist_wheel
running build
running build_py
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
running install_egg_info
running egg_info
writing top-level names to southridge.egg-info/top_level.txt
writing dependency_links to southridge.egg-info/dependency_links.txt
writing southridge.egg-info/PKG-INFO
reading manifest file 'southridge.egg-info/SOURCES.txt'
writing manifest file 'southridge.egg-info/SOURCES.txt'
removing 'build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info' (and everything under it)
Copying southridge.egg-info to build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info
running install_scripts
error: [('build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info', 'build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info', "[Errno 13] Permission denied: 'build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info'")]
```

I then tried to copy my folder into my `/home` folder and it worked here....
``` bash
spertus@MSI2019:/mnt/c/PROJECTS/DATA/sp$ sudo cp -r /mnt/c/PROJECTS/DATA/sp /home/sp
spertus@MSI2019:/home/sp/src$ sudo python3 setup.py bdist_wheel --universal
running bdist_wheel
running build
running build_py
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
running install_egg_info
running egg_info
writing southridge.egg-info/PKG-INFO
writing dependency_links to southridge.egg-info/dependency_links.txt
writing top-level names to southridge.egg-info/top_level.txt
reading manifest file 'southridge.egg-info/SOURCES.txt'
writing manifest file 'southridge.egg-info/SOURCES.txt'
removing 'build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info' (and everything under it)
Copying southridge.egg-info to build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info/WHEEL
```

Finally, i even tried to *re-install* my ubuntu distribution (tried 16 & 18) with better results ....

**Comment by Mimetis** [comment]
I made some others tests.

Another way to make it working, is to move everything to the folder inside the `Current User` folder.

I've tested it, and it worked as expected

``` bash 
$ cp -r /mnt/c/PROJECTS/DATA/sp /mnt/c/Users/spertus.EUROPE/Documents
$ cd /mnt/c/Users/spertus.EUROPE/Documents/sp/src
$ python setup.py bdist_wheel

...
# package generated as expected
...

```
I've checked the permissions on **every** folders (even the generated ones) in each directories, and the results are always the same

``` bash
$ cd /mnt/c/Users/spertus.EUROPE/Documents/sp/src
$ ls -l
drwxrwxrwx 1 spertus spertus 4096 Jul 25 09:11 build
-rwxrwxrwx 1 spertus spertus  364 Jul 25 09:11 setup.py
drwxrwxrwx 1 spertus spertus 4096 Jul 25 09:11 southridge
drwxrwxrwx 1 spertus spertus 4096 Jul 25 09:11 southridge.egg-info

$ cd /mnt/c/PROJECTS/DATA/sp/src
$ ls -l
 drwxrwxrwx 1 spertus spertus 4096 Jul 25 09:10 build
-rwxrwxrwx 1 spertus spertus  364 Jul 24 23:34 setup.py
drwxrwxrwx 1 spertus spertus 4096 Jul 24 23:45 southridge
drwxrwxrwx 1 spertus spertus 4096 Jul 25 09:10 southridge.egg-info
```
According to the error, it seems I have no permissions on `[Errno 13] Permission denied: 'build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info'"` 
But my permissions seems to be good.

``` bash
spertus@MSI2019:/mnt/c/PROJECTS/DATA/sp/src/build/bdist.linux-x86_64/wheel$ ls -l
total 0
drwxrwxrwx 1 spertus spertus 4096 Jul 25 09:10 southridge
drwxrwxrwx 1 spertus spertus 4096 Jul 25 09:10 southridge-0.9.dist-info
drwxrwxrwx 1 spertus spertus 4096 Jul 25 09:10 southridge-0.9.egg-info

spertus@MSI2019:/mnt/c/PROJECTS/OPENHACK/DATA/sp/src/build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info$ ls -l
total 0
-rwxrwxrwx 1 spertus spertus  1 Jul 25 09:10 dependency_links.txt
-rwxrwxrwx 1 spertus spertus 11 Jul 25 09:10 top_level.txt
```
Continuing my investigations...
Any advice would be greatly appreciated !

**Comment by Mimetis** [comment]
Adding a full control to **Authenticated Users** seems to resolve the issue

![image](https://user-images.githubusercontent.com/4592555/61855310-22a98680-aec0-11e9-99ae-017e47b91091.png)

After adding this new permissions, everything worked correctly, finally !

``` bash
spertus@MSI2019:/mnt/c/PROJECTS/DATA/SouthridgeProject/src$ python3 setup.py bdist_wheel --universal clean -a
running bdist_wheel
running build
running build_py
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/southridge
copying build/lib/southridge/cleanse.py -> build/bdist.linux-x86_64/wheel/southridge
copying build/lib/southridge/__init__.py -> build/bdist.linux-x86_64/wheel/southridge
running install_egg_info
running egg_info
writing top-level names to southridge.egg-info/top_level.txt
writing dependency_links to southridge.egg-info/dependency_links.txt
writing southridge.egg-info/PKG-INFO
reading manifest file 'southridge.egg-info/SOURCES.txt'
writing manifest file 'southridge.egg-info/SOURCES.txt'
Copying southridge.egg-info to build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info/WHEEL
running clean
removing 'build/lib' (and everything under it)
removing 'build/bdist.linux-x86_64' (and everything under it)
'build/scripts-3.5' does not exist -- can't clean it
removing 'build'
```

**Comment by blairg23** [comment]
> Adding a full control to **Authenticated Users** seems to resolve the issue
> 
> ![image](https://user-images.githubusercontent.com/4592555/61855310-22a98680-aec0-11e9-99ae-017e47b91091.png)
> 
> After adding this new permissions, everything worked correctly, finally !
> 
> ```shell
> spertus@MSI2019:/mnt/c/PROJECTS/DATA/SouthridgeProject/src$ python3 setup.py bdist_wheel --universal clean -a
> running bdist_wheel
> running build
> running build_py
> installing to build/bdist.linux-x86_64/wheel
> running install
> running install_lib
> creating build/bdist.linux-x86_64/wheel
> creating build/bdist.linux-x86_64/wheel/southridge
> copying build/lib/southridge/cleanse.py -> build/bdist.linux-x86_64/wheel/southridge
> copying build/lib/southridge/__init__.py -> build/bdist.linux-x86_64/wheel/southridge
> running install_egg_info
> running egg_info
> writing top-level names to southridge.egg-info/top_level.txt
> writing dependency_links to southridge.egg-info/dependency_links.txt
> writing southridge.egg-info/PKG-INFO
> reading manifest file 'southridge.egg-info/SOURCES.txt'
> writing manifest file 'southridge.egg-info/SOURCES.txt'
> Copying southridge.egg-info to build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info
> running install_scripts
> creating build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info/WHEEL
> running clean
> removing 'build/lib' (and everything under it)
> removing 'build/bdist.linux-x86_64' (and everything under it)
> 'build/scripts-3.5' does not exist -- can't clean it
> removing 'build'
> ```

Can confirm this fixes the issue using Ubuntu on the Linux for Windows Subsystem. I swear, I have had more issues with Windows permissions when developing using Windows... Hoped Linux for Windows Subsystem would make these issues go away...

**Comment by Aperion** [comment]
Another confirmation that the above fix indeed works. It worked fine under windows, but in the WSL instance in the same directory I would get the permission denied error. Adding full control to Authenticated Users solved. I encountered this issue while working with `scikit-build`

**Comment by glinders** [comment]
A workaround that worked for me on WSL is to add option '--bdist-dir ~/temp/bdistwheel'

**Comment by hellzxmaker** [comment]
When I go to add full permissions to my project dir, I am getting an error that I am not authorized to change the permissions of the egg-info file. Has anyone been able to work around this?

**Comment by shabraatgit** [comment]
Simply re-installed `wheel` package and the command worked fine. 
1. Install `wheel`
 `pip install wheel`
2. `python setup.py build bdist_wheel`

**Comment by shabraatgit** [comment]
> Adding a full control to **Authenticated Users** seems to resolve the issue
> 
> ![image](https://user-images.githubusercontent.com/4592555/61855310-22a98680-aec0-11e9-99ae-017e47b91091.png)
> 
> After adding this new permissions, everything worked correctly, finally !
> 
> ```shell
> spertus@MSI2019:/mnt/c/PROJECTS/DATA/SouthridgeProject/src$ python3 setup.py bdist_wheel --universal clean -a
> running bdist_wheel
> running build
> running build_py
> installing to build/bdist.linux-x86_64/wheel
> running install
> running install_lib
> creating build/bdist.linux-x86_64/wheel
> creating build/bdist.linux-x86_64/wheel/southridge
> copying build/lib/southridge/cleanse.py -> build/bdist.linux-x86_64/wheel/southridge
> copying build/lib/southridge/__init__.py -> build/bdist.linux-x86_64/wheel/southridge
> running install_egg_info
> running egg_info
> writing top-level names to southridge.egg-info/top_level.txt
> writing dependency_links to southridge.egg-info/dependency_links.txt
> writing southridge.egg-info/PKG-INFO
> reading manifest file 'southridge.egg-info/SOURCES.txt'
> writing manifest file 'southridge.egg-info/SOURCES.txt'
> Copying southridge.egg-info to build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info
> running install_scripts
> creating build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info/WHEEL
> running clean
> removing 'build/lib' (and everything under it)
> removing 'build/bdist.linux-x86_64' (and everything under it)
> 'build/scripts-3.5' does not exist -- can't clean it
> removing 'build'
> ```
Here is a simpler solution.
https://github.com/pypa/packaging-problems/issues/258#issuecomment-596395221

**Comment by SsmallWinds** [comment]
> Adding a full control to **Authenticated Users** seems to resolve the issue
> 
> ![image](https://user-images.githubusercontent.com/4592555/61855310-22a98680-aec0-11e9-99ae-017e47b91091.png)
> 
> After adding this new permissions, everything worked correctly, finally !
> 
> ```shell
> spertus@MSI2019:/mnt/c/PROJECTS/DATA/SouthridgeProject/src$ python3 setup.py bdist_wheel --universal clean -a
> running bdist_wheel
> running build
> running build_py
> installing to build/bdist.linux-x86_64/wheel
> running install
> running install_lib
> creating build/bdist.linux-x86_64/wheel
> creating build/bdist.linux-x86_64/wheel/southridge
> copying build/lib/southridge/cleanse.py -> build/bdist.linux-x86_64/wheel/southridge
> copying build/lib/southridge/__init__.py -> build/bdist.linux-x86_64/wheel/southridge
> running install_egg_info
> running egg_info
> writing top-level names to southridge.egg-info/top_level.txt
> writing dependency_links to southridge.egg-info/dependency_links.txt
> writing southridge.egg-info/PKG-INFO
> reading manifest file 'southridge.egg-info/SOURCES.txt'
> writing manifest file 'southridge.egg-info/SOURCES.txt'
> Copying southridge.egg-info to build/bdist.linux-x86_64/wheel/southridge-0.9.egg-info
> running install_scripts
> creating build/bdist.linux-x86_64/wheel/southridge-0.9.dist-info/WHEEL
> running clean
> removing 'build/lib' (and everything under it)
> removing 'build/bdist.linux-x86_64' (and everything under it)
> 'build/scripts-3.5' does not exist -- can't clean it
> removing 'build'
> ```

its works well!

**Comment by unfor19** [comment]
Embarrassed to admit, but my issue was - tar.gz file was occupied by 7-Zip, so `sdist` couldn't really re-create it 🤦🏻

**Comment by mwilcox12014** [comment]
Environment:  Windows 11, Conda 22.9.0, pip 22.2.2, python 3.9.13

Need modules from Allied Vision\Vimba_6.0 software
Run vimba setup.exe 

C: pip -v install . 
....
 '"'"''"'"''"'"' % ('"'"'C:\\Program Files\\Allied Vision\\Vimba_6.0\\VimbaPython\\Source\\setup.py'"'"',), "<pip-setuptools-caller>", "exec"))' bdist_wheel -d 'C:\Users\SOD\AppData\Local\Temp\pip-wheel-wjyrgxcs'
  cwd: C:\Program Files\Allied Vision\Vimba_6.0\VimbaPython\Source\
  Building wheel for VimbaPython (setup.py) ... error

Issue is Access denied on:  C:\Program Files\Allied Vision\Vimba_6.0\VimbaPython\Source 

As with others, Changing 'Users' permissions to 'Allow ' on Full Control fixed the problem.  

![image](https://user-images.githubusercontent.com/7051133/205161393-2f36c129-f9d5-44fd-be7b-1bf0c3c996ed.png)
